lib/_stream_readable.js

```diff
@@ -227,6 +227,15 @@ Readable.prototype._destroy = function(err, cb) {
   cb(err);
 };
 
+Readable.prototype[EE.captureRejectionSymbol] = function(err) {
+  // TODO(mcollina): remove the destroyed if once errorEmitted lands in
+  // Readable.
+  if (!this.destroyed) {
+    this.pause();
```

Comment:
I think the current consensus is that a `Readable` should flush out all it's data upon error?

Reply:
I think it depends on the use case and the consumer of the stream. New data should not be added to the readable buffer. Whether or not to read any remaining buffered data is consumer decision.

Reply:
Hm, should destroy with error always be pause then?

Reply:
It seems reasonable to me if it does not break anything on the ecosystem.

Reply:
It just seems strange to me that we would require extra logic here that does not usually apply. I’ll look into the current behavior.

Reply:
Changing this doesn't seem to be a big problem for streams. However, it is a huge problem for http2 which has lots of tests which asserts for me strange behaviour where e.g. `'end'` is emitted even when aborted or errored. Seems like a rather big undertaking to change. I also think http2 calls destroy instead of end in a few cases which makes things even harder.

My main concern here is not with the pause itself but the fact that it is different/inconsistent with the usual error handling flow. Honestly I would prefer to just make this a simple `this.destroy(err)` without condition or `pause()` just to make it easier to reason about. Unless there is a reason for pause here that I've missed?

Reply:
I've removed pause().

Reply:
I prefer not to remove the `if (destroyed)` because it's needed backporting. After this land, I'll open a follow-up PR to remove the if.